cmd/aws-k8s-agent/main.go

```diff
@@ -39,18 +37,17 @@ func _main() int {
 
 	log.Infof("Starting L-IPAMD %s  ...", version)
 
-	kubeClient, err := k8sapi.CreateKubeClient()
-	if err != nil {
-		log.Errorf("Failed to create client: %v", err)
+	//Check API Server Connectivity
+	if k8sapi.CheckAPIServerConnectivity() != nil{
```

Comment:
... why do this, rather than just wait until the first actual apiserver call to effectively do it for us?

(In particular, just because the apiserver was available _here_ doesn't mean it's _always_ going to be available during every other part of execution - so we still have to handle intermittent connectivity issues elsewhere.)

Reply:
Sure, a success here doesn't guarantee the call will go through a while later. This was mainly done to be in parity with the current CNI codeflow, where we check API server connectivity and essentially crash if the initial API Server connectivity check fails. Sort of helps to catch or highlight basic connectivity issues to API server as part of CNI bootstrap..

https://github.com/aws/amazon-vpc-cni-k8s/blob/master/pkg/k8sapi/discovery.go#L87